_resources/assets/scripts.js

```diff
@@ -0,0 +1,37 @@
+// Load and render Mermaid diagrams.
+window.addEventListener("load", () => {
+    if (document.querySelector('pre.mermaid')) {
+        const element = document.createElement('script')
+        element.src = 'https://unpkg.com/mermaid@8.0.0-alpha.6/dist/mermaid.min.js'
+        document.head.appendChild(element)
+        element.addEventListener("load", function() {
+            mermaid.mermaidAPI.initialize({
+                startOnLoad: false,
```

Comment:
I love the idea of defining a Gantt chart with mermaid, but the usage docs example looks a lot simpler than this: https://mermaid-js.github.io/mermaid/#/usage?id=include-mermaid-on-your-web-page
Could you explain what all that extra code is for?

Reply:
@sqs to confirm, but the solution in the docs was resulting in the height and width of the injected SVG being set to 0, and we were needing to do some manual tweaks to make it show.

Reply:
@felixfbecker:

1. That window.mermaid.init API is deprecated. The docs say: "Warning This type of integration is deprecated. Instead the preferred way of handling more complex integration is to use the mermaidAPI instead."
1. We only want to load the JavaScript if Mermaid is used on the page.
1. We want to let people customize the size with `data-rendered-width` (so it scrolls to the right if desired, instead of being tiny to fit the width) and `data-scroll-right` (so it's not scrolled all the way to the beginning of time).

Reply:
Could you add that as code comments? 🙂

Reply:
Yes, I'll add in these comments!

Reply:
@felixfbecker - I updated the file with comments as you requested. Let me know if I need to add any other clarifications!